src/util.rs

```diff
@@ -29,16 +29,22 @@ pub fn get_default_data_dir() -> PathBuf {
 }
 
 // Function to initialize the logger with the given data directory
-pub fn init_logger(data_dir: &PathBuf) -> Result<(), Box<dyn Error + Send + Sync>> {
+pub fn init_logger(data_dir: &PathBuf, log_level: &str ) -> Result<(), Box<dyn Error + Send + Sync>> {
     let log_file = data_dir.join("debug.log");
-
     let config = ConfigBuilder::new().build();
+    let level_filter = match log_level {
+        "error" => LevelFilter::Error,
+        "warn" => LevelFilter::Warn,
+        "info" => LevelFilter::Info,
+        "debug" => LevelFilter::Debug,
+        "trace" => LevelFilter::Trace,
+        _ => panic!("Invalid log level in config"),
+    };
+
     let log_writer = File::create(&log_file)?;
```

Comment:
I got an error here `Error: Os { code: 2, kind: NotFound, message: "No such file or directory” }` when the default dar dir is not existent, I wonder if we should test in `get_default_dar_dir()` if a data dir is already present, and in lack of it create a new one ?

Reply:
Made a push to fix this. Try it out?

Reply:
Yes I see the fix `fs::create_dir_all` it works.